tekstowo.py

```diff
@@ -0,0 +1,464 @@
+# Necessary imports
+from typing import Tuple, List, Dict
+from datetime import datetime
+from bs4 import BeautifulSoup
+
+import os
+import argparse
+import string
+import time
+import requests
+import langdetect
+
+# Parser config
+parser = argparse.ArgumentParser(description="Crawler and scraper dedicated to tekstowo.pl domain")
+parser.add_argument("--letter", 
+    "--ARTIST_LETTER", 
+    help="Choose a letter to scrape the lyrics from (default = Q)", 
+    default="Q", 
+    type=str)
+parser.add_argument("--save_progress", 
+    "--SAVE_PROGRESS", 
+    help="Choose the interval of creating a save_progress file", 
+    default=30, 
+    type=int)
+
+args = parser.parse_args()
+
+# Config
+ARTIST_LETTER: str = args.letter
+SAVE_PROGRESS: int = args.save_progress # interval of creating a save state file
+
+# TODO:
+# create functions: save_progress, load_progress, continue_cycle - DONE
+# decide how to save progress: either txt with specific formatting or a json file - DONE
+# progress bar?
+
+# Function collection
+def generate_timestamp() -> str:
+    """
+    Returns current system time.
+    """
+    # Get the current time
+    current_datetime = datetime.now()
+
+    # Format the date and time as a string
+    timestamp = current_datetime.strftime("%Y-%m-%d, %H:%M:%S")
+
+    return timestamp
+
+def processing_time(start_timestamp: datetime, end_timestamp: datetime) -> Tuple[int, int, int, int]:
+    """
+    Counts the time that has passed performing a given task.
+    """
+    processing_time = end_timestamp - start_timestamp
+    total_seconds = processing_time.total_seconds()
+
+    days = int(total_seconds // (3600 * 24))
+    hours = int((total_seconds // 3600) % 24)
+    minutes = int((total_seconds % 3600) // 60)
+    seconds = int(total_seconds % 60)
+
+    return days, hours, minutes, seconds
+
+def get_max_page_number(url: str) -> int:
+    """
+    Get the highest page number per given letter.
+    Used in cooperation with 'create_lut_pagination' 
+    and 'pages_per_letter' method.
+    """
+
+    max_page = 0
+
+    try:
+        req = requests.get(url, timeout=60)
+        req.raise_for_status()
+    
```

Comment:
A single space isn't need in the try-except block of code as it is one, continuous 'block entity'.

Reply:
Changed to no space in try-except block.